inginious/agent/docker_agent/__init__.py

```diff
@@ -441,15 +440,24 @@ async def create_student_container(self, parent_info, socket_id, environment_nam
                 return
 
             environment = self._containers[environment_type][environment_name]["id"]
-            runtime = self._containers[environment_type][environment_name]["runtime"]
+
+            if run_as_root:
+                runtime_name = {k for k in self._runtimes if self._runtimes[k].run_as_root}.pop()
+                runtime = self._runtimes[runtime_name].runtime
```

Comment:
Does this mean that you select a random runtime? Why can't we select one in `run_student` ?

Reply:
Indeed, this implementation forced to use kata runtime if run_as_root otherwise, it selected randomly a non-root runtime.
This, while working in the situation with only docker and kata runtimes, is indeed an issue.
Now, with last modifications, it forces Kata if run_as_root otherwise, it uses the runtime specified by arguments, as it was the case before this pull request.